src/engine/renderer/tr_shader.cpp

```diff
@@ -5239,35 +5239,57 @@ static void FinishStages()
 		? gl_shaderManager.getDeformShaderIndex( shader.deforms, shader.numDeforms )
 		: 0;
 
+	bool isOpaqueShader = false;
+
 	for ( size_t s = 0; s < numStages; s++ )
 	{
 		shaderStage_t *stage = &stages[ s ];
 
 		stage->deformIndex = deformIndex;
 
-		// We should cancel overbright if there is no light stage.
-		stage->cancelOverBright = shaderHasNoLight;
+		// SRC1 and DST0 are reset to zero in ParseStage (no blending).
+		bool isOpaque = !( stage->stateBits & ( GLS_SRCBLEND_BITS | GLS_DSTBLEND_BITS ) );
+
+		// A shader is not opaque if all stages are not opaques.
+		isOpaqueShader |= isOpaque;
```

Comment:
To clarify: is the intention to make the shader opaque when at least 1 stage is opaque, or if all stages are opaque?

Reply:
It is opaque if at least one stage is opaque.

Reply:
In fact the information for the whole shader is not used, but when processing a stage, we must know if there is nothing opaque between the background the shader will be painted over, and the current stage of the shader being processed.